src/adapter/src/optimize/index.rs

```diff
@@ -163,6 +164,9 @@ impl Optimize<Index> for Optimizer {
         };
         df_desc.export_index(self.exported_index_id, index_desc, on_desc.typ().clone());
 
+        // TODO(sdht0): consider other timelines such as `TimelineContext::TimestampIndependent`.
+        df_desc.timeline = Some(Timeline::EpochMilliseconds);
```

Comment:
Do we need to consider other timeline types? For now, we can just say that replica expiration only works for `Timeline::EpochMilliseconds` kind, and not configure the expiration otherwise.

Reply:
Only `EpochMilliseconds`, all others we can ignore.

Reply:
Can you expand on this a bit? How can we just blindly assign the `EpochMilliseconds` timeline?

Reply:
> Can you expand on this a bit? How can we just blindly assign the EpochMilliseconds timeline?

It's a good question. We have to figure out how to set the appropriate value here, perhaps by threading in the timeline.

Reply:
If you have the `GlobalId` and a handle to the `Coordinator`, then you should be able to use this method: https://github.com/MaterializeInc/materialize/blob/8d1128a98d1887a336d59b65d6238a755276796e/src/adapter/src/coord/timeline.rs#L411-L415

Reply:
Fixed